support condition branch in ao debug handler

Gasoonjia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #141612
* __->__ #141516

This diff introduced the supportive of condition statement into ao debug handler generation.

Most of code borrowed from ExecuTorch to avoid circle dependency issue.

Differential Revision: [D66270691](https://our.internmc.facebook.com/intern/diff/D66270691/)

Gasoonjia committed:
support condition branch in ao debug handler

This diff introduced the supportive of condition statement into ao debug handler generation.

Most of code borrowed from ExecuTorch to avoid circle dependency issue.

Differential Revision: [D66270691](https://our.internmc.facebook.com/intern/diff/D66270691/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "support condition branch in ao debug handler"


This diff introduced the supportive of condition statement into ao debug handler generation.

Most of code borrowed from ExecuTorch to avoid circle dependency issue.

Differential Revision: [D66270691](https://our.internmc.facebook.com/intern/diff/D66270691/)

[ghstack-poisoned]

Gasoonjia committed:
Update on "support condition branch in ao debug handler"


This diff introduced the supportive of condition statement into ao debug handler generation.

Most of code borrowed from ExecuTorch to avoid circle dependency issue.

Differential Revision: [D66270691](https://our.internmc.facebook.com/intern/diff/D66270691/)

[ghstack-poisoned]

jerryzh168 commented on torch/ao/quantization/pt2e/_numeric_debugger.py:
```diff
@@ -42,26 +43,50 @@ def generate_numeric_debug_handle(ep: ExportedProgram) -> None:
-    for node in ep.graph.nodes:
+
+    def _bfs_trace_graph_with_node_process(node_op: Callable) -> None:
+        nonlocal ep
```
nit: why not pass this as an argument?

Gasoonjia replied:
Just wanna make `_bfs_trace_graph_with_node_process` more generic

jerryzh168 replied:
looks like this takes an arg in the next PR

Gasoonjia committed:
Update on "support condition branch in ao debug handler"


This diff introduced the supportive of condition statement into ao debug handler generation.

Most of code borrowed from ExecuTorch to avoid circle dependency issue.

Differential Revision: [D66270691](https://our.internmc.facebook.com/intern/diff/D66270691/)

[ghstack-poisoned]
